stellar-dotnet-sdk/TransactionBuilder.cs

```diff
@@ -0,0 +1,160 @@
+using System;
+using System.Collections.Concurrent;
+using stellar_dotnet_sdk.xdr;
+
+namespace stellar_dotnet_sdk
+{
+    public class TransactionBuilder
+    {
+        private const uint BaseFee = 100;
+
+        private readonly BlockingCollection<Operation> _operations;
+        private readonly ITransactionBuilderAccount _sourceAccount;
+        private Memo _memo;
+        private TimeBounds _timeBounds;
+        private uint _fee;
+
+        /// <summary>
+        ///     Construct a new transaction builder.
+        /// </summary>
+        /// <param name="sourceAccount">
+        ///     The source account for this transaction. This account is the account
+        ///     who will use a sequence number. When build() is called, the account object's sequence number will be incremented.
+        /// </param>
+        public TransactionBuilder(ITransactionBuilderAccount sourceAccount)
+        {
+            _sourceAccount = sourceAccount ??
+                             throw new ArgumentNullException(nameof(sourceAccount), "sourceAccount cannot be null");
+            _operations = new BlockingCollection<Operation>();
+            _fee = BaseFee;
+        }
+
+        public static FeeBumpTransaction BuildFeeBumpTransaction(IAccountId feeSource, Transaction inner, long fee)
+        {
+            var innerOps = inner.Operations.Length;
+            var innerBaseFeeRate = inner.Fee / innerOps;
+
+            if (fee < innerBaseFeeRate)
+            {
+                throw new Exception($"Invalid fee, it should be at least {innerBaseFeeRate} stroops");
+            }
+
+            if (fee < BaseFee)
+            {
+                throw new Exception($"Invalid fee, it should be at least {BaseFee} stroops");
+            }
+
+            var feeBumpFee = fee * (innerOps + 1);
```

Comment:
missing overflow check

Reply:
Also add a Unit Test to cover overflow scenario.